snafu/config.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Tools for setting up config arguments."""
+import os
+from typing import Iterable, List
+import argparse
+import configargparse
+
+
+def check_file(file: str, perms: int = None) -> bool:
+    """
+    Check that the given file (relative or absolute dir) exists and has the given minimum permissions.
+    Given ``perms`` should be combination of ``os.R_OK``, ``os.W_OK`` and ``os.X_OK``. By default
+    will just check if the file is readable.
+    """
+
+    if perms is None:
+        perms = os.R_OK
+    perms |= os.F_OK
+    return os.access(os.path.abspath(file), perms)
+
+
+class ConfigArgument:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+
+class Config:
+    """
+    Class for managing config parameters.
+
+    Uses argparse and configargparse in the background
+    """
+
+    def __init__(self, tool_name: str):
+        self.config: argparse.Namespace = argparse.Namespace()
```

Comment:
Maybe a better name for this attribute? So its not ``config.config``?

Reply:
`config.params` ?

Reply:
Done deal.